front/components/assistant/conversation/input_bar/InputBar.tsx

```diff
@@ -161,44 +168,46 @@ export function AssistantInputBar({
     setContentFragmentBody(undefined);
   };
 
-  const onInputFileChange: InputBarContainerProps["onInputFileChange"] = async (
-    event
-  ) => {
-    const file = (event?.target as HTMLInputElement)?.files?.[0];
-    if (!file) return;
-    if (file.size > 10_000_000) {
-      sendNotification({
-        type: "error",
-        title: "File too large.",
-        description:
-          "PDF uploads are limited to 10Mb per file. Please consider uploading a smaller file.",
-      });
-      return;
-    }
-    const res = await handleFileUploadToText(file);
+  const onInputFileChange: InputBarContainerProps["onInputFileChange"] =
+    useCallback(
+      async (event) => {
+        const file = (event?.target as HTMLInputElement)?.files?.[0];
+        if (!file) return;
+        if (file.size > 10_000_000) {
+          sendNotification({
+            type: "error",
+            title: "File too large.",
+            description:
+              "PDF uploads are limited to 10Mb per file. Please consider uploading a smaller file.",
+          });
+          return;
+        }
+        const res = await handleFileUploadToText(file);
 
-    if (res.isErr()) {
-      sendNotification({
-        type: "error",
-        title: "Error uploading file.",
-        description: res.error.message,
-      });
-      return;
-    }
-    if (res.value.content.length > 1_000_000) {
-      // This error should pretty much never be triggered but it is a possible case, so here it is.
-      sendNotification({
-        type: "error",
-        title: "File too large.",
-        description:
-          "The extracted text from your PDF has more than 1 million characters. This will overflow the assistant context. Please consider uploading a smaller file.",
-      });
-      return;
-    }
+        if (res.isErr()) {
+          sendNotification({
+            type: "error",
+            title: "Error uploading file.",
+            description: res.error.message,
+          });
+          return;
+        }
+        if (res.value.content.length > 1_000_000) {
+          // This error should pretty much never be triggered but it is a possible case, so here it is.
+          sendNotification({
+            type: "error",
+            title: "File too large.",
+            description:
+              "The extracted text from your PDF has more than 1 million characters. This will overflow the assistant context. Please consider uploading a smaller file.",
+          });
+          return;
+        }
 
-    setContentFragmentFilename(res.value.title);
-    setContentFragmentBody(res.value.content);
-  };
+        setContentFragmentFilename(res.value.title);
+        setContentFragmentBody(res.value.content);
+      },
+      [sendNotification]
```

Comment:
curious that there's a dependency on sendNotification, given it's global context (and thus won't change)
react mystery for me

Reply:
Yeah in this case it does not make a lot of sense.